### PR TITLE
added binding support for rfc 5705

### PR DESCRIPTION
**Pull request by kelbyludwig**
I am working on a PR for pyopenssl that allows for exporting keying material and needed to expose this API.

### CODE REVIEW

**Comment by kelbyludwig** [comment]
https://github.com/pyca/pyopenssl/pull/686 will fail without these updates to cryptography. if there is a smarter way of tackling this just let me know!

**Comment by alex** [comment]
There's a few places where this deviates from our style guide: https://cryptography.io/en/latest/development/c-bindings/

Can you please fix those up?

(Also, for some reason we seem to be missing coverage, but that's not related to your patch. We'll need to fix that before this can be merged)

**Comment by kelbyludwig** [comment]
@alex Thanks for pointing that guide out. I think my PR is aligned to those guidelines now.

File: src/_cffi_src/openssl/cryptography.py [hunk 1]
```diff
@@ -25,6 +25,8 @@
 #include <windows.h>
 #endif
 
+#define CRYPTOGRAPHY_OPENSSL_101_OR_GREATER \
+    (OPENSSL_VERSION_NUMBER >= 0x1000100 && !CRYPTOGRAPHY_IS_LIBRESSL)
```
Comment (reaperhulk) [inline src/_cffi_src/openssl/cryptography.py]: We only support OpenSSL 1.0.1+ so this should always be true for OpenSSL. Additionally, it looks like `SSL_export_keying_material` is available in LibreSSL so I don't think you actually need to use a conditional binding at all here.

Comment (kelbyludwig) [reply src/_cffi_src/openssl/cryptography.py]: Makes sense. I assume that means I should remove the changes to `src/cryptography/hazmat/bindings/openssl/_conditional.py` as well?

Comment (reaperhulk) [reply src/_cffi_src/openssl/cryptography.py]: Yeah you should only need the function definition and none of the other boilerplate.

File: src/_cffi_src/openssl/ssl.py [hunk 2]
```diff
@@ -520,6 +523,7 @@
 static const long Cryptography_HAS_SSL_OP_NO_TICKET = 1;
 static const long Cryptography_HAS_SSL_SET_SSL_CTX = 1;
 static const long Cryptography_HAS_NEXTPROTONEG = 1;
+static const long Cryptography_HAS_KEYING_EXPORT = 1;
```
Comment (reaperhulk) [inline src/_cffi_src/openssl/ssl.py]: This (and the declaration) are also unnecessary now. Other than that this looks good!

Comment (kelbyludwig) [reply src/_cffi_src/openssl/ssl.py]: you beat me to it :)

**Comment by reaperhulk** [comment]
This LGTM now. Hopefully codecov will sort itself out once all the tests actually run 😄

**Comment by reaperhulk** [comment]
Thanks for the contribution!

**Pull request merged**